string name
-------
int16 success